botorch/models/deterministic.py

```diff
@@ -76,6 +76,18 @@ def posterior(
             values = values[..., output_indices]
         return DeterministicPosterior(values=values)
 
+    def _set_transformed_inputs(self) -> None:
+        r"""This is a no-op since DeterministicModel does not have a `train_inputs`
+        attribute and does not go through training.
+        """
+        pass  # pragma: no cover
+
+    def _revert_to_original_inputs(self) -> None:
+        r"""This is a no-op since DeterministicModel does not have a `train_inputs`
+        attribute and does not go through training.
+        """
+        pass  # pragma: no cover
```

Comment:
It would be good if one wouldn't have to overwrite this in subclasses that don't have a `train_inputs` attribute. Can we just modify the function in `Model` to be a nullup if there is no `train_inputs` attribute?

Reply:
Yeah, makes sense. I did that and added a warning so that this doesn't turn into another gotcha down the line.